src/acl/Options.h

```diff
@@ -13,51 +13,84 @@
 #include "sbuf/forward.h"
 
 #include <iosfwd>
-#include <map>
-#include <set>
+#include <vector>
 
-// After all same-name acl configuration lines are merged into one ACL:
-//   configuration = acl name type [option...] [[flag...] parameter...]
-//   option = -x[=value] | --name[=value]
-//   flag = option
+// After line continuation is handled by the preprocessor, an ACL object
+// configuration can be visualized as a sequence of same-name "acl ..." lines:
 //
-// Options and flags use the same syntax, but differ in scope and handling code:
-// * ACL options appear before all parameters and apply to all parameters.
-//   They are handled by ACL kids (or equivalent).
-// * Parameter flags may appear after some other parameters and apply only to
-//   the subsequent parameters (until they are overwritten by later flags).
-//   They are handled by ACLData kids.
-// ACL options parsing code skips and leaves leading parameter flags (if any)
-// for ACLData code to process.
+// L1: acl exampleA typeT parameter1 -i parameter2 parameter3
+// L2: acl exampleA typeT parameter4
+// L3: acl exampleA typeT -i -n parameter5 +i parameter6
+// L4: acl exampleA typeT -n parameter7
+//
+// There are two kinds of ACL options (a.k.a. flags):
+//
+// * Global (e.g., "-n"): Applies to all parameters regardless of where the
+//   option was discovered/parsed (e.g., "-n" on L3 affects parameter2 on L1).
+//   Declared by ACL class kids (or equivalent) via ACL::options().
+//
+// * Line (e.g., "-i"): Applies to the yet unparsed ACL parameters of the
+//   current "acl ..." line (e.g., "-i" on L1 has no effect on parameter4 on L2)
+//   Declared by ACLData class kids (or equivalent) via lineOptions().
+//
+// Here is the option:explicitly-affected-parameters map for the above exampleA:
+//   "-n": parameter1-7 (i.e. all parameters)
+//   "-i": parameter2, parameter3; parameter5
+//   "+i": parameter6
+//
+// The option name spelling determines the option kind and effect.
+// Both option kinds use the same general option configuration syntax:
+//   option = name[=value]
+// where "name" is option-specific spelling that looks like -x, +x, or --long
```

Comment:
Please use ABNF syntax like the original comment (sort of) did. It is easier for anyone familiar with syntax to interpret and validate correctness against.

```suggestion
//    option = ('-' | '+') (a-z)  [ '=' value]
//          \= '-' '-' name [ '=' value]
```


or, if we do the deprecation idea below it would become:

```suggestion
//    option = ('-' | '+') (a-z)
//          \= '-' '-' name [ '=' value]
```

Reply:
We should not follow the POSIX standard for command line syntax because this is not a command line, it is an optional flag on a config file directive.

IMO the option names being coded as beginning with the +/-/-- characters is an implementation optimization from the assumption that the option/flags cannot be merged (eg "-i+i-i+i" [no whitespace] is not valid). The config syntax has those characters as indicators reserved with a distinct meaning to the parser telling it what type of option to parse (short vs long), separate from what the option name is and means.

Reply:
That said, the update makes the current documentation clear and accurate. So am am not insisting on any further change here.